src/HttpAsyncClientDiscovery.php

```diff
@@ -24,7 +24,7 @@ public static function find()
         try {
             $asyncClient = static::findOneByType(HttpAsyncClient::class);
 
-            return new $asyncClient();
+            return static::instantiateClass($asyncClient);
```

Comment:
Sorry, I think I miss-read your comment. I meant that we do not need an *additional* try-catch here (at this line). 

> Do we need the instantiation within the try? Is it possible that a closure throws a DiscoveryFailedException?

Yes, we need the instantiation within a try-catch. Any weird exception could happen both with a FQCN (Class not found) and with the closure. The closure may throw a "VeryRandomException" that is not part of our code.